lib/Yare/Chain/Follower.hs

```diff
@@ -114,14 +135,6 @@ indexBlock addresses block !tip !state = (state', utxoUpdate)
       finality
       (look @Utxo state)
 
-  finality ∷ Utxo.Finality =
```

Comment:
This "finality" calculation has moved one level up the stack, such that chain follower can switch storage mode depending on it.